craft_application/services/project.py

```diff
@@ -0,0 +1,452 @@
+    _project_model: models.Project | None
+
+    def __init__(
+        self, app: AppMetadata, services: ServiceFactory, *, project_dir: pathlib.Path
+    ) -> None:
+        super().__init__(app, services)
+        self.__platforms = None
+        self.__partitions: list[str] | None = None
+        self.__project_file_path = None
+        self.__raw_project: dict[str, Any] | None = None
+        self._project_dir = project_dir
+        self._project_model = None
+        self._build_on: craft_platforms.DebianArchitecture | None = None
+        self._build_for: str | None = None
+        self._platform: str | None = None
+
+    @final
+    def configure(self, *, platform: str | None, build_for: str | None) -> None:
+        """Configure the prime project to render.
+
+        This method configures the settings of our prime project rendering -
+        that is, the one that will be used for the project that is cached.
+        """
+        self._build_on = craft_platforms.DebianArchitecture.from_host()
+        if self.is_configured:
+            raise RuntimeError("Project is already configured.")
+
+        platforms = self.get_platforms()
+        if platform and platform not in platforms:
+            raise errors.InvalidPlatformError(platform, list(platforms.keys()))
+
+        if platform and build_for:
+            self._platform = platform
+            self._build_for = (
+                build_for
+                if build_for == "all"
+                else craft_platforms.DebianArchitecture(build_for)
+            )
+            self.__is_configured = True
+            return
+
+        if not platform:
+            # If we don't have a platform, select the first platform that matches
+            # our build-on and build-for. If we don't have a build-for, select the
+            # first platform that matches our build-on.
+            for name, data in platforms.items():
+                if self._build_on.value not in data["build-on"]:
+                    continue
+                if build_for and build_for not in data["build-for"]:
+                    continue
+                platform = name
+                break
+            else:
+                if build_for:
+                    raise RuntimeError(
+                        f"Cannot generate a project that builds on "
+                        f"{self._build_on} and builds for {build_for}"
+                    )
+                # We won't be able to build in this case, but the project is
+                # still valid for non-lifecycle commands. Our prime project will
+                # just be the first item available.
+                platform = next(iter(platforms))
+                self._platform = platform
+                build_for = platforms[platform]["build-for"][0]
+                self._build_for = (
+                    build_for
+                    if build_for == "all"
+                    else craft_platforms.DebianArchitecture(build_for)
```

Comment:
This and line 115 below may need a try/catch.  Previously, these were caught by pydantic and raised a nice error.

craft-app 4:

```
Bad snapcraft.yaml content:
- 'amd66' is not a valid Debian architecture. (in field 'platforms.amd64.build-on')
```

craft-app 5:

```
snapcraft internal error: ValueError("'amd66' is not a valid DebianArchitecture")
Full execution log: '/home/developer/.local/state/snapcraft/log/snapcraft-20250320-105544.078333.log'
```

Reply:
That's a good catch. Mind if I do it in a separate PR?

Reply:
Yes, a separate PR would be great.

Reply:
#686